File: mac/AirClip/Core/DeviceRemovalNotice.swift
import Foundation

enum DeviceRemovalNotice {
    static let type = "device_removed"

    static func payload(targetDeviceId: String) -> [String: String] {
        [
            "type": type,
            "target_device_id": targetDeviceId,
        ]
    }

    static func targetsCurrentDevice(_ json: [String: Any], currentDeviceId: String?) -> Bool {
        guard
            json["type"] as? String == type,
            let targetDeviceId = json["target_device_id"] as? String,
            let currentDeviceId
        else {
            return false
        }
        return targetDeviceId == currentDeviceId
    }
}

File: android/app/src/main/java/com/airclip/airclip/device/DeviceRemovalNotice.kt
package com.airclip.airclip.device

object DeviceRemovalNotice {
    const val TYPE = "device_removed"
    const val TARGET_DEVICE_ID = "target_device_id"

    fun payload(targetDeviceId: String): Map<String, String> = mapOf(
        "type" to TYPE,
        TARGET_DEVICE_ID to targetDeviceId,
    )

    fun targetsCurrentDevice(message: Map<String, String>, currentDeviceId: String?): Boolean =
        message["type"] == TYPE &&
            currentDeviceId != null &&
            message[TARGET_DEVICE_ID] == currentDeviceId
}
